documentation/AuthServer.md

```diff
@@ -25,56 +53,110 @@ The document types used by the authorization server include:
 
-#### Client
+#### GPII Tokens
+
+| Option | Type | Description | Default |
+| ------ | ---- | ----------- | ------- |
+| `id` | String | The GPII token id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
+| `type` | String | The document type for storing GPII tokens. | The value must be set to "gpiiToken". |
+| `gpiiToken` | String | The GPII token. | None |
+| `userId` | String | The user id that this GPII token belongs to. | None |
+
+#### GPII App Installation Clients
 
 | Option | Type | Description | Default |
 | ------ | ---- | ----------- | ------- |
 | `id` | String | The client id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
-| `type` | String | The document type for storing client information. | The value must be set to "client". |
+| `type` | String | The document type for storing the client type information. | The value must be set to "gpiiAppInstallationClient". |
 | `name` | String | The client name. | None |
 | `oauth2ClientId` | String | The unique identifier issued to a registered OAuth2 client by the authorization server. | None |
 | `oauth2ClientSecret` | String | Confidential shared secret between the client and the authorization server, used to verify the identity of the client. | None |
-| `redirectUri` | String | The URL on client's site where users will be sent after authorization. | None |
+| `userId` | Boolean | The id of the user who creates this client.| None |
 
-#### GPII Tokens
+#### GPII App Installation Authorizations
 
 | Option | Type | Description | Default |
 | ------ | ---- | ----------- | ------- |
-| `id` | String | The GPII token id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
-| `type` | String | The document type for storing GPII tokens. | The value must be set to "gpiiToken". |
-| `gpiiToken` | String | The GPII token. | None |
-| `userId` | String | The user id that this GPII token belongs to. | None |
+| `id` | String | The GPII app installation authorizations id. Can be a UUID or any unique string. This value is saved into `_id` field in CouchDB/PouchDB. | None |
+| `type` | String | The document type for storing GPII app installation authorizationss. | The value must be set to "gpiiAppInstallationAuthorization". |
+| `clientId` | String | The client id that this token is assigned to. | None |
+| `gpiiToken` | String | The GPII token that this token record is associated with. | None |
+| `accessToken` | String | The access token used to retrieved the protected user preferences. | None |
+| `expiresIn` | Number | The number of seconds left before the access token becomes invalid. The default value is 3600. | 3600 |
+| `revoked` | Boolean | Whether this token has been revoked. | false |
+| `expired` | Boolean | Whether this token has expired, meaning `timestampCreated` + `expiresIn` < now(). | false |

```

Comment:
Thanks for thinking about this too. Are you suggesting to have the calculation of "expired" flags on the fly by couchdb views instead of having them stored in documents?

I was also debating whether or not to save "expired" values. The reason that I decided to save is, when we start to assign access tokens to GPII apps, this document will easily accumulate hundreds and thousands of access tokens. Without a flag to easily identify not-revoked and not-expired tokens by couchdb views, every time when a token needs to be requested/verified/retrieved, the calculation will need to happen across all access tokens which would likely introduce performance issues. 

Also, the calculation of "expired" is not as simple as `timestampCreated + expiresIn < now()` that can be performed in couchdb views. The calculation is via [this function](https://github.com/cindyli/universal/blob/GPII-2435/gpii/node_modules/gpii-oauth2/gpii-oauth2-utilities/src/OAuth2Utilities.js#L69), which means every time all access tokens will be fetched and calculated to find unexpired ones.

So, what's implemented at the moment is that, every time when an access token is requested by a GPII app, which is the time that all existing access tokens need to be gone through to find an unrevoked and unexpired token, the chance is taken to set the "expired" values, which will only update a small set of access tokens.

Let me know what you think. Thanks.